### PR TITLE
Fix typos and grammar

### PR DESCRIPTION
**Pull request by devisasari**

### CODE REVIEW

File: src/guide/essentials/template-syntax.md [hunk 1]
```diff
@@ -255,7 +255,7 @@ The above will be converted to `:[someattr]` in in-DOM templates. If your compon
 
 ### Modifiers
 
-Modifiers are special postfixes denoted by a dot, which indicate that a directive should be bound in some special way. For example, the `.prevent` modifier tells the `v-on` directive to call `event.preventDefault()` on the triggered event:
+Modifiers are special postfixes denoted by a dot, which indicates that a directive should be bound in some special way. For example, the `.prevent` modifier tells the `v-on` directive to call `event.preventDefault()` on the triggered event:
```
Comment (NataliaTepluhina) [inline src/guide/essentials/template-syntax.md:258]: Could you please include the fixes @skirtles-code mentioned in https://github.com/vuejs/docs/pull/2021? Here,  `indicate` is correct here, as the relevant noun is `modifiers`, not `dot`.

File: src/guide/introduction.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ If you find these concepts intimidating, don't worry! The tutorial and guide onl
...
 
-Despite the flexibility, the core knowledge about how Vue works is shared across all these use cases. Even if you are just a beginner now, the knowledge gained along the way will stay useful as you grow to tackle more ambitious goals in the future. If you are a veteran, you can pick the optimal way to leverage Vue based on the problems you are trying to solve, while retaining the same productivity. This is why we call Vue "The Progressive Framework": it's a framework that can grow with you and adapt to your needs.
+Despite the flexibility, the core knowledge about how Vue works is shared across all these use cases. Even if you are just a beginner now, the knowledge gained along the way will stay useful as you grow to tackle more ambitious goals in the future. If you are a veteran, you can pick the optimal way to leverage Vue based on the problems you are trying to solve while retaining the same productivity. This is why we call Vue "The Progressive Framework": it's a framework that can grow with you and adapt to your needs.
```
Comment (NataliaTepluhina) [inline src/guide/introduction.md:92]: As @skirtles-code mentioned in https://github.com/vuejs/docs/pull/2021#discussion_r996430469, previous version is more readable

File: src/guide/introduction.md [hunk 3]
```diff
@@ -209,7 +209,7 @@ If you are new to Vue, here's our general recommendation:
 
 - For production use:
 
-  - Go with Options API if you are not using build tools, or plan to use Vue primarily in low-complexity scenarios, e.g. progressive enhancement.
+  - Go with Options API if you are not using build tools, or plan to use Vue primarily in low-complexity scenarios, e.g., progressive enhancement.
```
Comment (NataliaTepluhina) [inline src/guide/introduction.md:212]: I don't think we need a comma here

**Review by NataliaTepluhina — COMMENTED** [review]
@devisasari thank you for your changes! I noticed you closed your previous PR and opened a lot of new ones again including this one. I will go ahead and close everything but this PR.

Could you please implement the mentioned changes to this branch? There is no need to close this PR and open a new one.

File: src/guide/introduction.md (hunk 3 above)
Comment (devisasari) [reply src/guide/introduction.md:212]: Yes, you're right

File: src/guide/introduction.md (hunk 2 above)
Comment (devisasari) [reply src/guide/introduction.md:92]: Yes, I agree with you

File: src/guide/essentials/template-syntax.md (hunk 1 above)
Comment (devisasari) [reply src/guide/essentials/template-syntax.md:258]: I am reviewing proposed changes in my pull request for the first time. Thank you.

**Review by bencodezen — APPROVED** [review]
Thanks for the detailed PR @devisasari and taking care of @NataliaTepluhina's suggestions. Approved from my end!

**Comment by devisasari** [comment]
> 

Thank you!

**Comment by bencodezen** [comment]
@NataliaTepluhina Going to leave this in your queue for you to approve and merge!

File: src/guide/introduction.md (hunk 2 above)
Comment (skirtles-code) [reply src/guide/introduction.md:92]: @devisasari We need you to make the change to your branch, we can't do it for you. The proposed change is below and you should be able to commit it to your branch by clicking on the button.

```suggestion
Despite the flexibility, the core knowledge about how Vue works is shared across all these use cases. Even if you are just a beginner now, the knowledge gained along the way will stay useful as you grow to tackle more ambitious goals in the future. If you are a veteran, you can pick the optimal way to leverage Vue based on the problems you are trying to solve, while retaining the same productivity. This is why we call Vue "The Progressive Framework": it's a framework that can grow with you and adapt to your needs.
```

File: src/guide/introduction.md (hunk 3 above)
Comment (skirtles-code) [reply src/guide/introduction.md:212]:
```suggestion
  - Go with Options API if you are not using build tools, or plan to use Vue primarily in low-complexity scenarios, e.g. progressive enhancement.
```

File: src/guide/essentials/template-syntax.md (hunk 1 above)
Comment (skirtles-code) [reply src/guide/essentials/template-syntax.md:258]:
```suggestion
Modifiers are special postfixes denoted by a dot, which indicate that a directive should be bound in some special way. For example, the `.prevent` modifier tells the `v-on` directive to call `event.preventDefault()` on the triggered event:
```

**Comment by NataliaTepluhina** [comment]
I went ahead and applied the suggestions @skirtles-code had added. Going to merge this PR now 🎉

**Review by NataliaTepluhina — APPROVED** [review]

**Pull request merged**